libs/cart/src/reducers/cart/cart.reducer.spec.ts

```diff
@@ -0,0 +1,286 @@
+    });
+  });
+
+  describe('when CartLoadFailureAction is triggered', () => {
+    const error = 'error message';
+    let result;
+    let state: CartState;
+
+    beforeEach(() => {
+      state = {
+        ...initialState,
+        loading: true,
+        errors: new Array('firstError')
+      }
+
+      const cartListLoadFailure = new DaffCartLoadFailure(error);
+
+      result = reducer(state, cartListLoadFailure);
+    });
+
+    it('should setloading to false', () => {
+      expect(result.loading).toEqual(false);
+    });
+
+    it('adds an error to state.errors', () => {
+      expect(result.errors.length).toEqual(2);
+    });
+  });
+
+  describe('when CartCreateAction is triggered', () => {
+    it('should setloading state to true', () => {
+      const cartCreateAction: DaffCartCreate = new DaffCartCreate();
+
+      const result = reducer(initialState, cartCreateAction);
+
+      expect(result.loading).toEqual(true);
+    });
+  });
+
+  describe('when CartCreateSuccessAction is triggered', () => {
+    let result;
+    let state: CartState;
+
+    beforeEach(() => {
+      state = {
+        ...initialState,
+        loading: true
+      }
+
+      const cartCreateSuccess = new DaffCartCreateSuccess({id: cart.id});
+
+      result = reducer(state, cartCreateSuccess);
+    });
+
+    it('should setcart from action.payload', () => {
+      expect(result.cart.id).toEqual(cart.id)
+    });
+
+    it('should setloading to false', () => {
+      expect(result.loading).toEqual(false);
+    });
+  });
+
+  describe('when CartCreateFailureAction is triggered', () => {
+    const error = 'error message';
+    let result;
+    let state: CartState;
+
+    beforeEach(() => {
+      state = {
+        ...initialState,
+        loading: true,
+        errors: new Array('firstError')
+      }
+
+      const cartCreateFailure = new DaffCartCreateFailure(error);
+
+      result = reducer(state, cartCreateFailure);
+    });
+
+    it('should setloading to false', () => {
+      expect(result.loading).toEqual(false);
+    });
+
+    it('adds an error to state.errors', () => {
+      expect(result.errors.length).toEqual(2);
+    });
+  });
+
+  describe('when AddToCartAction is triggered', () => {
+    const productId = 'productId';
+    const qty = 1;
+
+    it('should setloading state to true', () => {
+      const addToCartAction: DaffAddToCart = new DaffAddToCart({ productId, qty });
+
+      const result = reducer(initialState, addToCartAction);
+
+      expect(result.loading).toEqual(true);
+    });
+  });
+
+  describe('when AddToCartActionSuccess is triggered', () => {
+    let result;
+    let state: CartState;
+
+    beforeEach(() => {
+      const addToCartActionSuccess: DaffAddToCartSuccess = new DaffAddToCartSuccess(cart);
+      state = {
+        ...initialState,
+        loading: true
+      }
+
+      result = reducer(state, addToCartActionSuccess);
+    });
+
+    it('should setcart from action.payload', () => {
```

Comment:
A lot of typos like this in this file: `setcart`

Reply:
Changed in [`2eea542`](https://github.com/graycoreio/daffodil/pull/681/commits/2eea54244a04f5331df801d853e83d2dd2c74ab2).